closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -1070,12 +1084,12 @@ boolean isRestrictedProviderAllowed(String providerName) {
         }
 
         /**
-         * Get the provider name defined in provider construction method.
+         * Get the provider name without package name.
          *
-         * @param providerName provider name or provider with packages
-         * @return provider name defined in provider construction method
+         * @param providerName provider name with package name
+         * @return provider name without package name
          */
-        private static String getProvidersSimpleName(String providerName) {
+        private static String getProvidersNameWithoutPackage(String providerName) {
             if (providerName.equals("com.sun.security.sasl.Provider")) {
```

Comment:
I don't think we need or want to make SunSASL a special case any more.

Reply:
@keithc-ca We already released a version which suggested the client to set using SunSASL in their java.security provider list. So, that is why I did not remove this SunSASL provider mapping.

Reply:
With the change to `ProviderConfig.load()`, it should it not recognize both `SunSASL` and `com.sun.security.sasl.Provider`? I think so, which eliminates the need for a special case here.

Reply:
@keithc-ca 

If we set `RestrictedSecurity.NSS.140-2.jce.provider.N=com.sun.security.sasl.Provider,` this works. Because in the ServiceLoader, before the provider initialized, it will use the package + class name to check, and due to we use the package + class name in the java.security provider list, the check will pass.

But if we set `RestrictedSecurity.NSS.140-2.jce.provider.N=SunSASL`, this won't work. Because in the ServiceLoader, before the provider initialized, it will use the class name to check, and due to we use the provider name in the java.security provider list, and for SunSASL, its provider name is different with its class name(after removing the package), so it won't pass the check in ServiceLoader. 

For the non-java.base providers, the check happens in the ServiceLoader using its package + class name. So, if the provider name is different with its class name(after removing the package). We have to set its package + class name in the java.security RestrictedSecurity provider list.

Reply:
In retrospect, it was a mistake to try to support `RestrictedSecurity.NSS.140-2.jce.provider.N=SunSASL`.